File: bench/py/26_dict_iterate.py
# iterate over a dictionary's items. Checksum sums values (order-independent).
import sys
scale = int(sys.argv[1]) if len(sys.argv) > 1 else 1

M = 100000
d = {}
for i in range(M):
    d[i] = i * 2

R = 5 * scale
s = 0

for k in range(R):
    for key, val in d.items():
        s += val
    s = s % 1000000007

print("result:", s)

File: bench/cpp/26_dict_iterate.cpp
/* SPDX-License-Identifier: BSD-2-Clause */
/* Faithful C++ of bench/my/26_dict_iterate.my: foreach over a dict's <key,
 * value> pairs, summing values (order-independent checksum). */
#include "bench.h"
#include <unordered_map>

int main(int argc, char **argv)
{
    long scale = bench_scale(argc, argv);

    const long M = 100000;
    std::unordered_map<long, long> d;
    for (long i = 0; i < M; i++)
        d[i] = i * 2;

    long R = 5L * scale;
    long s = 0;
    for (long k = 0; k < R; k++) {
        for (const auto &kv : d)
            s += kv.second;
        s = s % 1000000007L;
    }

    printf("result: %ld\n", s);
    return 0;
}
